x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -247,6 +252,75 @@ export class DataRecognizer {
     return configs.find((i) => i.module.id === id);
   }
 
+  // called externally by an endpoint
+  public async findIndexMatches(
+    moduleId: string,
+    size: number = 15
+  ): Promise<RecognizeModuleResultDataView[]> {
+    const config = await this._findConfig(moduleId);
+    const matchingDataViews: RecognizeModuleResultDataView[] = [];
+
+    if (config?.module.query === undefined) {
+      return matchingDataViews;
+    }
+
+    try {
+      const promises: Array<
+        Promise<SearchResponse<unknown, Record<string, AggregationsAggregate>>>
+      > = [];
+      // Keep track of which data view title is associated with which promise
+      const dataViewTitleMap: Record<string, Promise<any>> = {};
+      const dataViewsMap: Record<string, { id: string; name: string | undefined }> = {};
+      const idsWithTitle = await this._dataViewsService.getIdsWithTitle();
+
+      idsWithTitle.forEach(({ id, title, name }) => {
+        dataViewsMap[title] = { id, name };
+        const promise = this._client.asCurrentUser.search(
+          {
+            index: title, // title is index pattern we can search by e.g. "filebeat-*"
+            size: 0,
+            body: {
+              query: config?.module.query,
+            },
+          },
+          { maxRetries: 0 }
+        );
+        dataViewTitleMap[title] = promise;
+        promises.push(promise);
+      });
+
+      const response = await Promise.all(promises);
+
+      const responseMap: Record<string, number> = Object.keys(dataViewTitleMap).reduce(
```

Comment:
you can't guarantee the order of the keys in an object, so these may not match up with the response array indices.

Reply:
Removed in c95eeaac6e88ebed1292d54ceeea5a6babbc65d1